### PR TITLE
Collection level role config

### PR DESCRIPTION
**Pull request by Idokah**
Get the collection role config from env (need to set them through the secret-manger / app config) 

example for the json that we are expecting in `process.env.ROLE_CONFIG`: 

```
collectionLevelConfig : [
  {
    "id":"collectionName1"
    "readPolicies": ["OWNER","BACKEND_CODE"],
    "writePolicies": ["OWNER","BACKEND_CODE"]
  },
  {
    "id":"collectionName2"
    "readPolicies": ["OWNER","BACKEND_CODE", "VISITOR"],
    "writePolicies": ["OWNER","BACKEND_CODE", "VISITOR"]
  }
]

```

### CODE REVIEW

File: packages/external-db-config/lib/readers/authorization_config_reader.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { checkRequiredKeys, isJson, EMPTY_ROLE_CONFIG } = require('../utils/config_utils')
+
+class AuthorizationConfigReader {
+  constructor() {
+  }
+
+  async readConfig() {
+    const { ROLE_CONFIG } = process.env
+    const { collectionLevelConfig } = isJson(ROLE_CONFIG) ? JSON.parse(ROLE_CONFIG) : EMPTY_ROLE_CONFIG
```
Comment (noam-almog) [inline packages/external-db-config/lib/readers/authorization_config_reader.js:9]: what happens in case the json is not parsing or contain bad data ?

**Comment by noam-almog** [comment]
do we plan to include app info representation of the auth data in this PR too ?

File: packages/external-db-config/lib/readers/authorization_config_reader.js (hunk 1 above)
Comment (Idokah) [reply packages/external-db-config/lib/readers/authorization_config_reader.js:9]: Empty role config.
Default permissions for all the collections

Comment (noam-almog) [reply packages/external-db-config/lib/readers/authorization_config_reader.js:9]: user can provide invalid json and not be aware of that.
what i mean is a valid json, but an invalid data.
we need to validate it as well and inform the user of that.

**Comment by Idokah** [comment]
> do we plan to include app info representation of the auth data in this PR too ?

I think I'll do it in another PR. I don't want this PR to be bigger.

File: packages/external-db-config/lib/readers/authorization_config_reader.js (hunk 1 above)
Comment (Idokah) [reply packages/external-db-config/lib/readers/authorization_config_reader.js:9]: WDYT that we need to validate for example? 
that the JSON is in the right format (`id`,` readPolicies`,` writePolicies`) ?

**Comment by noam-almog** [comment]
yes, validate the format

**Pull request merged**